src/app/cats/cats-detail-view/cats-detail-view.component.ts

```diff
@@ -31,5 +53,6 @@ export class CatsDetailViewComponent implements OnInit {
   onFormClose(success: boolean) {
     this.adoptionFormActive = false;
     this.showConfirmMsg = success;
+    this.cat.adopted = true;
```

Comment:
nie to miejesce, ten kod jest bezwarunkowo wywołany na zamknięciu, tymsamym ustawia rezerwację nieważne czy faktycznie była.